pkg/collector/corechecks/system/gpu/nvidia/cpuMetricsSender.go

```diff
@@ -0,0 +1,69 @@
+package nvidia
+
+import (
+	"errors"
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+type cpuMetricSender struct {
+	cpusRegex     *regexp.Regexp
+	cpuUsageRegex *regexp.Regexp
+}
+
+func (cpuMetricSender *cpuMetricSender) Init() error {
+	// List of CPUs and their usage/frequency, e.g. 2%@102,1%@102,0%@102,0%@102,off,off,off,off
+	regex, err := regexp.Compile(`CPU\s*\[((?:.,?)+)]`)
+	if err != nil {
+		return err
+	}
+	cpuMetricSender.cpusRegex = regex
+
+	regex, err = regexp.Compile(`(?P<cpuUsage>\d+)%@(?P<cpuFreq>\d+)|off`)
+	if err != nil {
+		return err
+	}
+	cpuMetricSender.cpuUsageRegex = regex
+
+	return nil
+}
```

Comment:
Since the regexp are static, this could be done in a `init` function (with a test checking the `Compile` succeeded). Same for all senders.

This way we remove the need to the struct, the interface and calling the init manually.

Reply:
We don't need the `*MetricSender` struct and manually calling the init.

```
var (
        cpuRegexp *regexp.Regexp
        cpuUsageRegexp *regexp.Regexp
)

func init() {
	// List of CPUs and their usage/frequency, e.g. 2%@102,1%@102,0%@102,0%@102,off,off,off,off
	cpuRegexp, _ := regexp.Compile(`CPU\s*\[((?:.,?)+)]`)
	cpuUsageRegexp, _ := regexp.Compile(`(?P<cpuUsage>\d+)%@(?P<cpuFreq>\d+)|off`)
}
```

Reply:
I guess it's a lot of overhead for nothing. We might add it in the future when we need it. Unless we're planning big change in the near future, keeping it simple as possible is better IMO.

But the PR works as it, so I'll let you decide :+1: .